Number_types/include/CGAL/GMP/Gmpq_type.h

```diff
@@ -228,6 +228,12 @@ class Gmpq
   double to_double() const noexcept;
   Sign sign() const noexcept;
 
+  bool is_zero() const
+  {
+    const __mpz_struct* zn = mpq_numref(mpq());
+    return mpn_zero_p(zn->_mp_d, zn->_mp_size);
```

Comment:
Why not use the same code as mpq_class, i.e. `mpq_sgn(mpq())==0`, or `mpz_size(mpq_numref(mpq()))==0`? Or if you want to use internals, you only need to check if `zn->_mp_size` is 0.